app/src/context/projects-context.tsx

```diff
@@ -28,34 +32,67 @@ export function ProjectsProvider({children}: {children: ReactNode}) {
   const [projects, setProjects] = useState<ProjectExtended[]>([]);
 
   useEffect(() => {
-    const init = async () => {
-      const localProjects = await getProjectsDB();
+    initProjects();
+  }, []);
+
+  /**
+   * Initializes the application by synchronizing local and remote projects.
+   * First, it fetches projects from the local database and sets them in the project state,
+   * then it synchronizes with remote projects to update the local state and database as needed.
+   *
+   * @async
+   * @function init
+   * @returns {Promise<void>} Resolves when initialization is complete.
+   */
+  const initProjects = async () => {
+    await syncLocalProjects();
```

Comment:
Aren't you going to have race conditions since you are using `useState` setters inside these async functions. 

E.g. you might get 

- syncLocalProjects
- setProjects(local result) queued
- syncRemoteProjects
- setProjects(remote result) queued 
- setProjects(remote result) executed
- setProjects(local result) executed

the result being the local would overwrite the remote, or vice versa.

Might need to instead have the async functions return results, then do a deterministic merge between them, and set it once.